modules/world/objects/agent.cpp

```diff
@@ -80,6 +80,16 @@ void Agent::ExecutionPlan(const float &dt) {
                             history_.back().first);
 }
 
+void Agent::addTrajectoryStep(const StateActionPair& state_action_pair){
```

Comment:
can you elaborate on this function?

Reply:
This function is intended to add the state retrieved from Carla to the state history matrix. And in the behavior part, the last state of an agent is retrieved as the current state for planning, I found similar implementation in `Agent::Execute`.